client/src/components/PlaceCard.tsx

```diff
@@ -1,167 +1,169 @@
-import AddIcon from '@mui/icons-material/Add';
-import RemoveIcon from '@mui/icons-material/Remove';
-import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
-import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
+import React, { useState, useContext } from "react";
+import AddIcon from "@mui/icons-material/Add";
+import RemoveIcon from "@mui/icons-material/Remove";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
 import { Button } from "@mui/material";
-import React from "react";
 import noImage from "../assets/noImage.png";
-import { PlaceCardProps, PlaceCardState } from "../types/PlaceCardType";
-import { FaCircle, FaCircleInfo } from "react-icons/fa6";
-import { TripContext } from '../contexts/TripContext';
-import { TripData } from '../dataObjects/TripData';
-import { PlaceData } from '../dataObjects/PlaceData';
-export class PlaceCard extends React.Component<PlaceCardProps, PlaceCardState>{
-		static contextType = TripContext;
-		declare context: React.ContextType<typeof TripContext>
-	
-    state: PlaceCardState = {
-      showDetails: false,
-      phone: "",
-      website: "",
-      hours: [],
-      date: new Date().getDay()
+import { PlaceCardProps } from "../types/PlaceCardType";
+import { TripContext } from "../contexts/TripContext";
+
+export default function PlaceCard({
+  place,
+  isResult,
+  mapRef,
+}: PlaceCardProps): React.ReactElement {
+  const [showDetails, setShowDetails] = useState<boolean>(false);
+  const [phone, setPhone] = useState<string>("");
+  const [website, setWebsite] = useState<string>("");
+  const [hours, setHours] = useState<string[]>([]);
+
+  const date: number = new Date().getDay();
+
+  const { addPlace, removePlace } = useContext(TripContext);
+
+  const showDetailsHandler = () => {
+    setShowDetails(true);
+    const map = mapRef.current;
+    if (!map || !place.place_id) {
+      console.log("Map or place ID is undefined");
```

Comment:
Should we have some sort of error handling here besides just a console printout?